test/models/thesis_test.rb

```diff
@@ -572,4 +551,168 @@ class ThesisTest < ActiveSupport::TestCase
+    thesis.metadata_complete = false
+    thesis.update(thesis.as_json)
+    assert_equal 'Not ready for publication', thesis.publication_status
+  end
+
+  test 'Setting issues_found sets status to "Not ready for publication"' do
+    thesis = theses(:publication_review)
+    assert_equal 'Publication review', thesis.publication_status
+    thesis.issues_found = true
+    thesis.update(thesis.as_json)
+    assert_equal 'Not ready for publication', thesis.publication_status
+  end
+
+  test 'Unsetting author graduation_confirmed sets status to "Not ready for publication"' do
+    thesis = theses(:publication_review)
+    assert_equal 'Publication review', thesis.publication_status
+    author = thesis.authors.first
+    author.graduation_confirmed = false
+    author.save
+    thesis.reload
+    assert_equal false, thesis.authors_graduated?
+    assert_equal 'Not ready for publication', thesis.publication_status
+  end
+
+  test 'Adding an active hold sets status to "Not ready for publication"' do
+    thesis = theses(:publication_review)
+    assert_equal 'Publication review', thesis.publication_status
+    assert_equal 1, thesis.holds.count
+    hold = thesis.holds.first
+    hold.status = 'active'
+    hold.save
+    thesis.reload
+    assert_equal true, thesis.active_holds?
+    assert_equal 'Not ready for publication', thesis.publication_status
+  end
+
+  test 'Setting an existing hold to "expired" will set the thesis status back to "Not ready for publication"' do
+    thesis = theses(:publication_review)
+    assert_equal 'Publication review', thesis.publication_status
+    assert_equal 1, thesis.holds.count
+    hold = thesis.holds.first
+    hold.status = 'expired'
+    hold.save
+    thesis.reload
+    assert_equal true, thesis.active_holds?
+    assert_equal 'Not ready for publication', thesis.publication_status
+  end
+
+  test 'Setting an existing hold to "released" can put the thesis into "Publication review" status' do
+    thesis = theses(:publication_review_except_hold)
+    assert_equal 'Not ready for publication', thesis.publication_status
+    assert_equal 1, thesis.holds.count
+    hold = thesis.holds.first
+    hold.status = 'released'
+    hold.save
+    thesis.reload
+    assert_equal 'Publication review', thesis.publication_status
+  end
+
+  test 'Adding a new hold will set the thesis status back to "Not ready for publication"' do
+    thesis = theses(:publication_review)
+    assert_equal 'Publication review', thesis.publication_status
+    hold = Hold.new({
+      "thesis" => thesis,
+      "date_requested" => "2021-01-03",
+      "date_start" => "2021-01-01",
+      "date_end" => "2021-04-01",
+      "hold_source" => HoldSource.first,
+      "status" => "active",
+    })
+    assert_equal true, hold.valid?
+    hold.save
+    thesis.reload
+    assert_equal 'Not ready for publication', thesis.publication_status
+  end
+
+  test 'Adding a new hold when a thesis is in "Pending publication" will change nothing' do
```

Comment:
Thoughts on whether these last two checks should also be done on a status of "published"?

Reply:
Yeah, I was wondering about that; I also wasn't sure whether tests relating to "published" should be handled when that method gets written. I can write them now while I'm thinking about it.